### PR TITLE
Update the dev manual a bit

### PR DESCRIPTION
**Pull request by fingolfin**

### CODE REVIEW

File: doc/dev/kernel.xml [hunk 1]
```diff
@@ -374,60 +375,105 @@ which does exactly this.
...
- SET_ELM_PLIST( l, 3, x3 );
- CHANGED_BAG( l );
- SET_LEN_PLIST(l, 3);
- CHANGED_BAG( l );
- return l;
- }
+Obj FuncFoo1(Obj self, Obj x)
+{
+    Obj x2, x3;
+    Obj l = NEW_PLIST( T_PLIST, 3 );
```
Comment (wucas) [inline doc/dev/kernel.xml]: I hereby remind you of your own suggestion:
```suggestion
    Obj list = NEW_PLIST( T_PLIST, 3 );
```
And for the rest as well.

**Comment by DominikBernhardt** [comment]
@fingolfin Just  a reminder: You wanted to add `static` in a few lines.

File: doc/dev/kernel.xml (hunk 1 above)
Comment (fingolfin) [reply doc/dev/kernel.xml]: Thanks :)

**Comment by fingolfin** [comment]
Updated with the various remarks plus some more addressed.

**Pull request merged**